prime-router/src/main/kotlin/Report.kt

```diff
@@ -200,6 +211,40 @@ class Report {
         return Report(schema, Table.create(columns), fromThisReport("deidentify"))
     }
 
+    // takes the data in the existing report and synthesizes different data from it
+    // the goal is to allow us to take real data in, move it around and scramble it so it's
+    // not able to point back to the actual records
+    fun synthesizeData(
+        synthesizeStrategies: Map<String, SynthesizeStrategy> = emptyMap(),
+        targetState: String? = null,
+        targetCounty: String? = null
+    ): Report {
+        val columns = schema.elements.map {
+            val synthesizedColumn = synthesizeStrategies[it.name]?.let { strategy ->
+                // look in the mapping parameter passed in for the current element
+                when (strategy) {
+                    // examine the synthesizeStrategy for the field
+                    // can be one of three values right now:
+                    // empty column, shuffle column, pass through column untouched
+                    SynthesizeStrategy.SHUFFLE -> {
+                        val shuffledValues = table.column(it.name).asStringColumn().shuffled()
+                        StringColumn.create(it.name, shuffledValues)
+                    }
+                    SynthesizeStrategy.FAKE -> {
+                        // generate random faked data for the column passed in
+                        buildFakedColumn(it.name, it, targetState, targetCounty)
```

Comment:
So, just so I'm clear, this generation of fake data is separate from the --input-fake option?
Put another way, supposed I were to use both `--input-fake`, and `--synthesize` - in that case, the system would generate a fake Report, then here, some of those fake data would get overwritten with new fake data, true?   "Even faker date", eh?  ;)     

That seems fine, I just want to make sure I'm understanding how it works.

Reply:
That is correct. This is a different step from `--input-fake`.